app/main.py

```diff
@@ -1,6 +1,27 @@
+import os
+import requests
+
+
+API_KEY = os.getenv("API_KEY")
+URL = "http://api.weatherapi.com/v1/current.json"
+CITY = "Paris"
+
+
 def get_weather() -> None:
-    # write your code here
-    pass
+    if not API_KEY:
+        raise ValueError("API_KEY is not set in environment variables")
+
+    response = requests.get(URL, params={"key": API_KEY, "q": CITY})
+
+    if response.status_code == 200:
+        weather_data = response.json()
+        current = weather_data["current"]
+        location = weather_data["location"]
+        print(f"Weather for city {location["name"]}, {location["country"]}: ")
+        print(f"Temperature: {current['temp_c']} Celsius")
+        print(f"Condition: {current['condition']['text']}")
+    else:
+        print("Failed to get weather data")
```

Comment:
Why Use a try Block Here?
Network Errors: The requests.get call can fail due to network issues, timeouts, or invalid URLs. Wrapping it in a try block ensures that these errors are caught and handled gracefully rather than causing your program to crash.

HTTP Errors: If the API returns an error response (e.g., 4xx or 5xx status codes), using response.raise_for_status() will raise an exception. This exception is caught by the except block to provide a meaningful error message.

JSON Parsing Errors: The response.json() call can fail if the response is not valid JSON. Wrapping this in a try block allows you to catch ValueError and handle it appropriately.

Missing Keys: The code assumes that certain keys will always be present in the JSON response. If the API response format changes or if the data is missing, this can raise a KeyError, which you can catch and handle to avoid crashes.

Unforeseen Errors: Catching a generic Exception ensures that any other unexpected errors are also handled, providing robustness to your function.